### PR TITLE
Necropolis IPCs

### PR DESCRIPTION
**Pull request by ParadoxSpace**
They can now be necro contractors. They are not meant to be corp reps.

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
!review

**Comment by Geevies** [comment]
think you should fine a mechanical solution to them not being corp reps

**Comment by mikomyazaki** [comment]
> They can now be necro contractors. They are not meant to be corp reps.

Enforce this mechanically. Create a subtype of corp rep for Necropolis with a different species blacklist.

e.g.
/datum/job/representative/necropolis
    blacklisted_species = list("Aut'akh Unathi", "Baseline Frame", "Hephaestus G1 Industrial Frame", "Hephaestus G2 Industrial Frame", "Xion Industrial Frame", "Zeng-Hu Mobility Frame", "Bishop Accessory Frame", "Shell Frame")


Also someone should change how that works, it's really messy >_<

**Comment by ParadoxSpace** [comment]
someone's gonna need to help me with this

File: code/game/jobs/faction/faction.dm [hunk 1]
```diff
@@ -25,7 +26,11 @@
 	. = list()
 
 	for (var/path in allowed_role_types)
-		. += SSjobs.type_occupations[path]
+		var/datum/job/role = SSjobs.type_occupations[path]
+		if(LAZYACCESS(job_species_blacklist, role.title))
+			role = SSjobs.type_occupations[path]
```
Comment (mikomyazaki) [inline code/game/jobs/faction/faction.dm]: This line actually isn't needed. 
```suggestion

```

**Pull request merged**